src/ReflectionFunction.php

```diff
@@ -18,25 +18,46 @@
 /**
  * AST-based reflection for function
  */
-class ReflectionFunction extends BaseReflectionFunction
+class ReflectionFunction extends BaseReflectionFunction implements IReflection
 {
     use ReflectionFunctionLikeTrait, InternalPropertiesEmulationTrait;
 
     /**
+     * Name of the function
+     *
+     * @var string|\Closure
+     */
+    private $functionName;
```

Comment:
Any benefits of remembering user-provided `$functionName` VS using `\ReflectionFunction::getName()`?

Reply:
`parent::getName()` is not initialized in all cases. Less to special case.

Reply:
:+1: :memo: (Note to self:) Verify test cases!!!

Reply:
The one case that wasn't being tested was when the function hadn't been loaded by PHP. Added this for both classes and functions.